### ISSUE TITLE
agda-mode: "cannot refine" inside instance argument even though term to be refined typechecks there

### ISSUE THREAD

**Issue by rntz** [agda/agda#2653]
I have a hole I'd like to fill with a term, and if I do so manually it typechecks, but using C-c C-r produces a "cannot refine" error (and using C-c C-SPC results in a more verbose error):

```agda
data Food : Set where
  apple : Food
  instance banana : Food

auto : {{f : Food}} -> Food
auto {{x}} = x

-- This works ok:
is-fine : Food
is-fine = auto {{apple}}

-- but if I try to refine below, it says "cannot refine":
cannot-refine : Food
cannot-refine = auto {{{!apple!}}}
```

**Comment by xekoukou** [comment]
This works for me and I am using a version that is higher that 2.5.2.
Maybe you can update and see if it works.

**Comment by andreasabel** [comment]
Same here, I think this is fixed on 2.5.3.
